pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleValueMultiColumnReaderWriter.java

```diff
@@ -0,0 +1,207 @@
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.io.readerwriter.impl;
+
+import com.linkedin.pinot.core.io.reader.impl.FixedByteSingleValueMultiColReader;
+import com.linkedin.pinot.core.io.readerwriter.BaseSingleValueMultiColumnReaderWriter;
+import com.linkedin.pinot.core.io.readerwriter.PinotDataBufferMemoryManager;
+import com.linkedin.pinot.core.io.writer.impl.FixedByteSingleValueMultiColWriter;
+import com.linkedin.pinot.core.segment.memory.PinotDataBuffer;
+import java.io.IOException;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Extension of {@link BaseSingleValueMultiColumnReaderWriter} that fully implements a
+ * SingleValue MultiColumn reader and writer for fixed size columns.
+ *
+ * This implementation currently does not support reading with context, but can be enhanced to do so.
+ */
+public class FixedByteSingleValueMultiColumnReaderWriter extends BaseSingleValueMultiColumnReaderWriter {
+
+  private List<FixedByteSingleValueMultiColWriter> _writers;
+  private List<FixedByteSingleValueMultiColReader> _readers;
+  private final int _numRowsPerChunk;
+
+  private final int[] _columnSizesInBytes;
+  private final PinotDataBufferMemoryManager _memoryManager;
+  private String _allocationContext;
+  private final long _chunkSizeInBytes;
+  private int _capacityInRows;
+  private List<PinotDataBuffer> _dataBuffers;
+  private int _numColumns;
+
+  /**
+   * Constructor for the class.
+   *
+   * @param numRowsPerChunk Number of rows per chunk of data buffer.
+   * @param columnSizesInBytes Int array containing size of columns in bytes.
+   * @param memoryManager Memory manager to be used for allocating memory.
+   * @param allocationContext Allocation context (for debugging).
+   */
+  public FixedByteSingleValueMultiColumnReaderWriter(int numRowsPerChunk, int[] columnSizesInBytes,
+      PinotDataBufferMemoryManager memoryManager, String allocationContext) {
+
+    _numRowsPerChunk = numRowsPerChunk;
+    _columnSizesInBytes = columnSizesInBytes;
+    _memoryManager = memoryManager;
+    _allocationContext = allocationContext;
+
+    _writers = new ArrayList<>();
+    _readers = new ArrayList<>();
+    _dataBuffers = new ArrayList<>();
+
+    _capacityInRows = 0;
+    int rowSizeInBytes = 0;
+    for (int columnSizesInByte : columnSizesInBytes) {
+      rowSizeInBytes += columnSizesInByte;
+    }
+
+    _numColumns = _columnSizesInBytes.length;
+    _chunkSizeInBytes = rowSizeInBytes * numRowsPerChunk;
+  }
+
+  @Override
+  public int getInt(int row, int column) {
+    int rowInChunk = row % _numRowsPerChunk;
```

Comment:
can we set the num RowsPerChunk to be a power of 2 and do bit operation to compute modulo?

Reply:
In this PR it is always a power of 2. Perhaps we can assert power of 2 and use bit operations?

Reply:
yes. let's do that. can be part of another PR